### PR TITLE
Feature/ CC-0114 show quotation list in admin

### PR DESCRIPTION
**Pull request by Double-Twelve**

### CODE REVIEW

File: src/cart_items/cartItem.service.ts [hunk 1]
```diff
@@ -31,6 +31,33 @@ export class CartItemService {
...
+    if (user_id) {
+      optionalQuery["user_id"] = user_id;
+    }
+    const sorting = sort ? { [sort]: desc } : { createdAt: -1 };
+    const cartItems = await this.cartItemModel
+      .find({
+        $and: [{ isDeleted: false }],
+        $or: [optionalQuery],
+      })
+      .sort(sorting as { [key: string]: SortOrder | { $meta: "textScore" } })
```
Comment (ShawnWang86b) [inline src/cart_items/cartItem.service.ts:50]: what's this sort mean?

File: src/cart_items/dto/findAll-cartItem.dto.ts [hunk 2]
```diff
@@ -6,3 +6,16 @@ export class FindAllCartItemDto extends PaginationQueryDto {
   @IsOptional()
   user_id: string;
 }
+
+export class FindAllCartItemByAdminDto extends PaginationQueryDto {
+  @IsString()
+  @IsOptional()
+  user_id: string;
+
+  @IsString()
```
Comment (ShawnWang86b) [inline src/cart_items/dto/findAll-cartItem.dto.ts:15]: what are sort and number

**Review by ShawnWang86b — APPROVED** [review]

File: src/cart_items/dto/findAll-cartItem.dto.ts [hunk 3]
```diff
@@ -6,3 +6,16 @@ export class FindAllCartItemDto extends PaginationQueryDto {
   @IsOptional()
   user_id: string;
 }
+
+export class FindAllCartItemByAdminDto extends PaginationQueryDto {
```
Comment (zhuocun) [inline src/cart_items/dto/findAll-cartItem.dto.ts:10]: Is it necessary to add the limit and offset in dto? :)

File: src/cart_items/cartItem.service.ts [hunk 4]
```diff
@@ -31,6 +31,33 @@ export class CartItemService {
       .exec();
   }
 
+  async findAllByAdmin(findAllCartItem: FindAllCartItemByAdminDto): Promise<{
```
Comment (zhuocun) [inline src/cart_items/cartItem.service.ts:34]: Is the name "findAll" controversial? :) I remember that 锤姐 mentioned it in our group the other day.

File: src/cart_items/cartItem.service.ts (hunk 1 above)
Comment (Double-Twelve) [reply src/cart_items/cartItem.service.ts:50]: sort can determine the sort order of the returned data
the reference is https://mongoosejs.com/docs/api.html#query_Query-sort

File: src/cart_items/dto/findAll-cartItem.dto.ts (hunk 2 above)
Comment (Double-Twelve) [reply src/cart_items/dto/findAll-cartItem.dto.ts:15]: I am not clear your question and could you explain more?

File: src/cart_items/dto/findAll-cartItem.dto.ts (hunk 3 above)
Comment (Double-Twelve) [reply src/cart_items/dto/findAll-cartItem.dto.ts:10]: Removed

File: src/cart_items/cartItem.service.ts (hunk 4 above)
Comment (Double-Twelve) [reply src/cart_items/cartItem.service.ts:34]: Change to _findCartItemListByAdmin_

**Review by Cecilia0109 — APPROVED** [review]
LGTM！

**Review by zhuocun — APPROVED** [review]
Good job!

**Pull request merged**